### PR TITLE
VRF terminal unit PLR limit warnings under low heating loads

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
VRF terminal units show bad PLR limit warnings under low heating loads when no load air flow rates are different from air flow used during heating. This fixes issue #5925. Differences in results are unexpected since this typically affects non-autosized simulations and those that use 0 for the no load air flow rate.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [x] Code style (parentheses padding, variable names)
 - [x] Functional code review (it has to work!)
 - [x] If defect, results of running current develop vs this branch should exhibit the fix
 - [x] CI status: all green or justified
 - [x] Performance: CI Linux results include performance check -- verify this
 - [x] Unit Test(s)
 - C++ checks:
   - [x] Argument types
   - [x] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes n/a
 - [x] Documentation changes in place
 - [x] Changed docs build successfully

### CODE REVIEW

File: src/EnergyPlus/HVACVariableRefrigerantFlow.cc [hunk 1]
```diff
@@ -5958,29 +5958,41 @@ namespace HVACVariableRefrigerantFlow {
 			return;
 		}
 
-		// Get result when DX coil is off
-		PartLoadRatio = 0.0;
+		// Get result when DX coil is operating at the minimum PLR (1E-20) if not otherwise specified
+		PartLoadRatio = VRFTU( VRFTUNum ).MinOperatingPLR;
```
Comment (rraustad) [inline src/EnergyPlus/HVACVariableRefrigerantFlow.cc]: OK, so what was happening. This use specified 0 no load flow rate in the TU object. The model first checks PLR = 0 then PLR = 1. For a 0 no load air flow input, the output at PLR=0 is 0 watts, but the output at PLR = 1E-20 is much greater than 0 (model discontinuity). The model iterates to find the PLR to meet the load. When heating loads are small, at the operating heating air flow rate (PLR>0), the fan heat can exceed the load. Remember that the VRF TU operates the entire time step so at full load flow there will be fan heat easily greater than a few watts. The fix was to use a minimum PLR, which has been asked for by manufacturers. I set the default to 1.0E-20 until an input is provided so users can select their own value.

QUESTION: should I add the input for min PLR now?

Comment (rraustad) [inline src/EnergyPlus/HVACVariableRefrigerantFlow.cc]: OK, so what was happening. This use specified 0 no load flow rate in the TU object. The model first checks PLR = 0 then PLR = 1. For a 0 no load air flow input, the output at PLR=0 is 0 watts, but the output at PLR = 1E-20 is much greater than 0 (model discontinuity). The model iterates to find the PLR to meet the load. When heating loads are small, at the operating heating air flow rate (PLR>0), the fan heat can exceed the load. Remember that the VRF TU operates the entire time step so at full load flow there will be fan heat easily greater than a few watts. The fix was to use a minimum PLR, which has been asked for by manufacturers. I set the default to 1.0E-20 until an input is provided so users can select their own value.

QUESTION: should I add the input for min PLR now?

Comment (rraustad) [inline src/EnergyPlus/HVACVariableRefrigerantFlow.cc:5962]: OK, so what was happening. This user specified 0 no load flow rate in the TU object. The model first checks PLR = 0 then PLR = 1. For a 0 no load air flow input, the output at PLR=0 is 0 watts, but the output at PLR = 1E-20 is much greater than 0 (model discontinuity). The model iterates to find the PLR to meet the load. When heating loads are small, at the operating heating air flow rate (PLR>0), the fan heat can exceed the load. Remember that the VRF TU operates the entire time step so at full load flow there will be fan heat easily greater than a few watts. The fix was to use a minimum PLR, which has been asked for by manufacturers. I set the default to 1.0E-20 until an input is provided so users can select their own value.

QUESTION: should I add the input for min PLR now? Anytime the output at min PLR exceeds the load the TU will be off and operate at the no load air flow rate.

File: src/EnergyPlus/HVACVariableRefrigerantFlow.cc [hunk 2]
```diff
@@ -6058,10 +6070,10 @@ namespace HVACVariableRefrigerantFlow {
 
 					if ( VRF( VRFCond ).VRFAlgorithmTypeNum == AlgorithmTypeFluidTCtrl ) {
 					// Algorithm Type: VRF model based on physics, appliable for Fluid Temperature Control
-						VRFTU( VRFTUNum ).CalcVRF_FluidTCtrl( VRFTUNum, FirstHVACIteration, TempMaxPLR, TempOutput, OnOffAirFlowRatio );
+						VRFTU( VRFTUNum ).CalcVRF_FluidTCtrl( VRFTUNum, FirstHVACIteration, TempMinPLR, TempOutput, OnOffAirFlowRatio );
 					} else {
 					// Algorithm Type: VRF model based on system curve
-						CalcVRF( VRFTUNum, FirstHVACIteration, TempMaxPLR, TempOutput, OnOffAirFlowRatio );
+						CalcVRF( VRFTUNum, FirstHVACIteration, TempMinPLR, TempOutput, OnOffAirFlowRatio );
```
Comment (rraustad) [inline src/EnergyPlus/HVACVariableRefrigerantFlow.cc:6076]: This is the code where PLR limits are tightened. It appears the code used the wrong variable.

File: src/EnergyPlus/HVACVariableRefrigerantFlow.cc [hunk 3]
```diff
@@ -6079,10 +6091,10 @@ namespace HVACVariableRefrigerantFlow {
 
 							if ( VRF( VRFCond ).VRFAlgorithmTypeNum == AlgorithmTypeFluidTCtrl ) {
 							// Algorithm Type: VRF model based on physics, appliable for Fluid Temperature Control
-								VRFTU( VRFTUNum ).CalcVRF_FluidTCtrl( VRFTUNum, FirstHVACIteration, TempMinPLR, TempOutput, OnOffAirFlowRatio );
+								VRFTU( VRFTUNum ).CalcVRF_FluidTCtrl( VRFTUNum, FirstHVACIteration, PartLoadRatio, TempOutput, OnOffAirFlowRatio );
 							} else {
 							// Algorithm Type: VRF model based on system curve
-								CalcVRF( VRFTUNum, FirstHVACIteration, TempMinPLR, TempOutput, OnOffAirFlowRatio );
+								CalcVRF( VRFTUNum, FirstHVACIteration, PartLoadRatio, TempOutput, OnOffAirFlowRatio );
```
Comment (rraustad) [inline src/EnergyPlus/HVACVariableRefrigerantFlow.cc:6097]: This code calculates the output at the PLR last used when the iteration limit was exceeded. When the RegulaFalsi flag = -1, the function returns the last PLR and the output is at this PLR not the lower limit passed to RegulaFalsi.

File: src/EnergyPlus/HVACVariableRefrigerantFlow.cc [hunk 4]
```diff
@@ -5958,29 +5958,41 @@ namespace HVACVariableRefrigerantFlow {
...
-			VRFTU( VRFTUNum ).CalcVRF_FluidTCtrl( VRFTUNum, FirstHVACIteration, 0.0, NoCompOutput, OnOffAirFlowRatio );
+			VRFTU( VRFTUNum ).CalcVRF_FluidTCtrl( VRFTUNum, FirstHVACIteration, PartLoadRatio, NoCompOutput, OnOffAirFlowRatio );
 		} else {
 		// Algorithm Type: VRF model based on system curve
-			CalcVRF( VRFTUNum, FirstHVACIteration, 0.0, NoCompOutput, OnOffAirFlowRatio );
+			CalcVRF( VRFTUNum, FirstHVACIteration, PartLoadRatio, NoCompOutput, OnOffAirFlowRatio );
 		}
 
 		if ( VRFCoolingMode && HRHeatingMode ) {
 			// IF the system is in cooling mode, but the terminal unit requests heating (heat recovery)
```
Comment (rraustad) [inline src/EnergyPlus/HVACVariableRefrigerantFlow.cc:5973]: In the section below, if the output at min PLR exceeds the load the system is turned off and then operates at the no load air flow rate if using constant fan operating mode.

**Comment by rraustad** [comment]
Defect file before and after. Error file shows in V8.6 release:

    ** Warning ** ZoneHVAC:TerminalUnit:VariableRefrigerantFlow "CORE ZONE VRF TERMINAL UNIT"
    **   ~~~   ** Terminal unit part-load ratio calculation failed: PLR limits of 0 to 1 exceeded
    **   ~~~   ** Please fill out a bug report and forward to the EnergyPlus support group.
    **   ~~~   **  Environment=San Francisco Intl Ap CA USA TMY3 WMO#=724940, at Simulation time=01/13 06:30 - 06:35

Results comparison. Note that when small heating load exists, TU fan reverts to no load air flow rate, in this case 0. No warnings in error file regarding PLR limits.

5925-comparison

File: src/EnergyPlus/HVACVariableRefrigerantFlow.cc (hunk 1 above)
Comment (mjwitte) [reply src/EnergyPlus/HVACVariableRefrigerantFlow.cc:5962]: @rraustad If mfrs asked for a min PLR input, then I guess that's useful.  But if the fix here solve the issue, maybe it's not worth the trouble.

**Comment by mjwitte** [comment]
@rraustad I've tested this and it looks good - solves the problem with only tiny changes in annual results.  Does this change need to be mentioned in the engineering reference?

**Comment by rraustad** [comment]
Eng Ref updated.

**Pull request merged**